### DISCUSSION TITLE
Custom errors set in Per-VIRTUAL_HOST default configuration

### DISCUSSION THREAD

**Discussion by Zeychiel** [nginx-proxy/nginx-proxy#1977] [Q&A]
Hi,

I am intercepting all errors, from all containers (bar.com and foo.bar.com), to display custom pages.
I am able to do this using a `/etc/nginx/vhost.d/default` file : 

> proxy_intercept_errors on;
> 
> error_page    400 401 402 403 404 500 502 503 504   /404.html;
> location = /404.html {
>     root   /usr/share/nginx/html;
>     internal; # I tried without this too
> }

The custom error is displayed, but the problem is I cannot get the style and images of it doing : 

> src="/img/image.gif"
> or even
> src="image.gif"

The browser still seem to try to get the static files from the location of the website

> https://foo.bar.com/img/style.css
> not from  /usr/share/nginx/html

I tried several methods but can't get this to work, any help would be well appreciated :D
Thanks,

**Comment by wader** [comment]
Maybe embed all resources into the page? if you have a dedicated url for the error resources you could use `<base>`.

**Comment by Zeychiel** [comment]
I don't have any dedicated URL for the errors ; and embedding can be a solution, but isn't there a way to do it the 'normal' way ?

**Comment by wader** [comment]
You could try something like:

```
location = /error-resources/ {
    root /usr/share/nginx/html;
}
```

And then use `<img src="/error-resources/img/image.gif">`

But then i think you will intercept all "/error-resources/..." request for the proxied web servers. The `internal` directive is to tell nginx that the location is only valid for `error_page` etc and not accessible by http clients over network, in your case images and css files can't be internal as the browser will fetch them
